libsigopt/aux/utils.py

```diff
@@ -0,0 +1,20 @@
+# Copyright © 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache License 2.0
+import numbers
+
+import numpy
+
+
+def is_integer(num):
+  if isinstance(num, bool):
+    return False
+  return isinstance(num, (int | numbers.Integral))
+
+
+def is_number(x):
+  if isinstance(x, bool):
+    return False
+  if isinstance(x, float) and not numpy.isfinite(x):
```

Comment:
Not a super huge deal, but `math.isinf()` and `math.isnan()` should do the same thing without requiring more than the python standard library, just to keep a dependency from bleeding out unnecessarily.